src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/ReflectionFunction.cs

```diff
@@ -156,21 +158,27 @@ internal TexlFunction GetTexlFunction()
                 };
             }
 
-            return new CustomTexlFunction(info.Name, info.RetType, info.ParamTypes)
+            return new CustomTexlFunction(info.Name, FunctionCategories.UserDefined, info.RetType, info.ParamTypes)
             {
                 _impl = (runtimeConfig, args, cancellationToken) => InvokeAsync(runtimeConfig, args, cancellationToken),
                 LamdaParamMask = info.LamdaParamMask,
             };
         }
 
+        internal string GetFunctionName()
+        {
+            var info = Scan();
+            return info.Name;
+        }
+
         public FormulaValue Invoke(IServiceProvider serviceProvider, FormulaValue[] args)
         {
             return InvokeAsync(serviceProvider, args, CancellationToken.None).Result;
         }
 
         public async Task<FormulaValue> InvokeAsync(IServiceProvider serviceProvider, FormulaValue[] args, CancellationToken cancellationToken)
         {
-            Scan();
+            var info = Scan();

```

Comment:
> var info = Scan(); [](http://example.com/codeflow?start=11&length=19)

Why do we again need to Scan() if we _info is already populated in ctor?

Reply:
because this class have 2 construtors and one of it set _info is null.

Reply:
What if we call Scan() there?

Reply:
the old code use just Scan() because scan will update _info. But now we change _info to readonly so it cannot be modified in scan. so my newest update will need pass scan to variable and use that variable.